src/allmydata/node.py

```diff
@@ -213,6 +233,87 @@ def get_config(self, section, option, default=_None, boolean=False):
                 )
             return default
 
+    def get_config_from_file(self, name, required=False):
+        """Get the (string) contents of a config file, or None if the file
+        did not exist. If required=True, raise an exception rather than
+        returning None. Any leading or trailing whitespace will be stripped
+        from the data."""
+        fn = os.path.join(self._basedir, name)
+        try:
+            return fileutil.read(fn).strip()
+        except EnvironmentError:
+            if not required:
+                return None
+            raise
```

Comment:
Couple uncovered things here but I see this code is merely moving verbatim from elsewhere in this file.  So this is fine.

Similar comments for the uncovered code in `get_or_create_private_config` and `get_private_config`.